ad_type_classification/ad_creative_body_processor.py

```diff
@@ -0,0 +1,55 @@
+import itertools
+import unicodedata
+
+import nltk
+from nltk.tokenize import word_tokenize
+from nltk.corpus import stopwords
+from nltk.stem.porter import PorterStemmer
+
+def is_punctuation(glyph):
+    unicode_category = unicodedata.category(glyph)
+    return unicode_category and (unicode_category[0] == 'P')
+
+
+class AdCreativeBodyProcessor:
+    """Class to encapsulate ad creative body processing logic."""
+
+    def __init__(self):
+        nltk.download('stopwords')
```

Comment:
I *think* these downloads are only needed once per virtual env. I might be wrong.

Reply:
That's true, but they only download files if needed and only add a few seconds to execution time. I prefer to leave them to prevent silent failure and/or someone else having to figure this out later